### DISCUSSION TITLE
Missing error messages

### DISCUSSION THREAD

**Discussion by cbemeijer** [OpenModelica/OpenModelica#10311] [Q&A]
Hello, I am completely new to Open Modelica. To get started I was trying to follow the instructions in the [introduction](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/introduction.html#system-overview). 

What is frustrating me is that i do not get any error messages besides grammar errors from the OMShell. This is frustrating to me for two reasons. First of all it makes it harder to follow the tutorial, but more importantly I have learned to rely on error messages, especially when learning a new language. 

I have tried to install the latest version, and the versions seem to match up now with the one given in the introduction. (except my version is 64 bit) This doesn't seem to make a difference. The getErrorString() function returns empty strings even when i would expect it (edit: not) to.

Can someone please tell me what I am doing wrong?

**Comment by perost** [comment]
OMShell isn't very user friendly, so unless there's some specific reason why you want to use it I'd suggest trying out [OMEdit](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omedit.html) instead.

**Reply by cbemeijer** [reply]
My reasons for using OMShell don't really get more specific than that is was in the introduction on the OpenModelica website. I have been doing exercises in the OMWebbook. It's ok. What I am primarily interested in is to make models of individual machines, partly based on statistical data. Combining these machines into a model is mainly the job of my proces engineer colleagues, though it is something I also want to be able to do. I don't know if OM already has tools for this. I am also interested in the interoperability with C and (mainly)Python. There is some material on OM out there, but most seems to be more aimed at process engineers than developers. I would be grateful if someone could point me in the right direction.

**Reply by perost** [reply]
Perhaps we need to update the introduction, the main tool for users is OMEdit so having examples with OMShell in the introduction is perhaps not the best. OMEdit is an IDE for developing and using Modelica models, so if you want to develop your own models I would very much recommend trying out that tool instead.

As for interoperability with C and Python it depends on what you mean. Modelica has builtin support for calling C and Fortran functions through the [external function interface](https://specification.modelica.org/master/functions.html#external-function-interface), and I guess it would be possible to call Python code through a C wrapper too. We also have [OMPython](https://github.com/OpenModelica/OMPython) for interacting with OpenModelica in Python.

**Reply by cbemeijer** [reply]
Thank you very much!

**Reply by casella** [reply]
In any case, if there is some specific sequence that you type in OMShell and produces unexpected results, please open a ticket about that, and we can follow it up.

**Comment by gkaf89** [comment]
It is not an issue only in the OMShell, OMCompiler could benefit from more helpful error messages too.

Accidentally, I tried to [generate code from a file with extension `.om` instead of `.mo`](https://stackoverflow.com/questions/76929375/cannot-generate-code-for-a-simple-example-in-openmodelica). The compiler provided a fairly generic error message,
```
Error processing file: HelloWorld.om
# Error encountered! Exiting...
# Please check the error message and the flags.

Execution failed!
``` 
which wasn't very helpful. Something informing about an unknown file extensions would be much better.

I am new to OpenModelica, but should some issues be created for improving error reporting?

**Reply by perost** [reply]
> I am new to OpenModelica, but should some issues be created for improving error reporting?

Sure, giving good error messages can be hard, but we always welcome suggestions on how to improve them.
